### PR TITLE
Add degrees alias to rad2deg.

### PR DESCRIPTION
**Pull request by hameerabbasi**

### CODE REVIEW

**Comment by mruberry** [comment]
Unfortunately you'll need to rebase because https://github.com/pytorch/pytorch/pull/42802 introduced a new way of adding aliases. It should be pretty simple, though.

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -405,7 +405,11 @@ def method_tests():
         ('ceil', (S, S, S), NO_ARGS, '', (True,)),
         ('ceil', (), NO_ARGS, 'scalar', (True,)),
         ('rad2deg', (S, S, S), NO_ARGS),
+        ('rad2deg_', (S, S, S), NO_ARGS),
         ('deg2rad', (S, S, S), NO_ARGS),
+        ('deg2rad_', (S, S, S), NO_ARGS),
+        ('degrees', dont_convert((S, S, S)), NO_ARGS),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:411]: You can remove degrees from here, which should solve some test failures.

File: aten/src/ATen/native/UnaryOps.cpp [hunk 2]
```diff
@@ -142,7 +146,7 @@ Tensor& atan_(Tensor& self) { return unary_op_impl_(self, at::atan_out); }
 // Complex inputs to abs and angle return float results by default.
 // abs and angle, in both NumPy and C++, returns a float result when given a
 // complex input. This makes sense mathematically since the absolute value
-// and angle of a complex number has no imaginary part.
+// and angle of a complex number has no imaginary part.d
```
Comment (mruberry) [inline aten/src/ATen/native/UnaryOps.cpp:149]: 'd'

File: aten/src/ATen/native/UnaryOps.cpp [hunk 3]
```diff
@@ -130,6 +130,10 @@ Tensor& deg2rad_out(Tensor& result, const Tensor& self) {
 Tensor deg2rad(const Tensor& self) { return unary_op_impl(self, at::deg2rad_out); }
 Tensor& deg2rad_(Tensor& self) { return unary_op_impl_(self, at::deg2rad_out); }
 
+Tensor degrees(const Tensor& self) { return at::native::rad2deg(self); }
```
Comment (mruberry) [inline aten/src/ATen/native/UnaryOps.cpp:133]: Comment documenting this is an alias of rad2deg. Might as well put these entries immediately under the rad2deg entries, too.

File: aten/src/ATen/native/native_functions.yaml [hunk 4]
```diff
@@ -2423,6 +2423,18 @@
 - func: deg2rad.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
   supports_named_tensor: True
 
+- func: degrees(Tensor self) -> Tensor
```
Comment (mruberry) [inline aten/src/ATen/native/native_functions.yaml:2426]: Add a comment that degrees is an alias for rad2deg and move these under the rad2deg entries

File: torch/_torch_docs.py [hunk 5]
```diff
@@ -5416,6 +5416,13 @@ def merge_dicts(*dicts):
 
 """.format(**common_args))
 
+add_docstr(torch.degrees,
+           r"""
+degrees(input, out=None) -> Tensor
```
Comment (mruberry) [inline torch/_torch_docs.py:5421]: input, *, out=None

File: torch/testing/_internal/common_methods_invocations.py [hunk 6]
```diff
@@ -405,7 +405,9 @@ def method_tests():
         ('ceil', (S, S, S), NO_ARGS, '', (True,)),
         ('ceil', (), NO_ARGS, 'scalar', (True,)),
         ('rad2deg', (S, S, S), NO_ARGS),
+        ('rad2deg_', (S, S, S), NO_ARGS),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:408]: You can remove this and the the deg2rad_ entry. I don't think method_tests supports inplace ops, and we don't need to add tests for these inplace operators here.

**Comment by hameerabbasi** [comment]
The Windows CI builds have an internal compiler error, possibly related, but this PR isn't the source of the issue I think.

The Linux CI jobs fail with:

```python
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/test/test_jit.py", line 6330, in test_math_ops
    checkMathWrap(op, 1)
  File "/var/lib/jenkins/workspace/test/test_jit.py", line 6256, in checkMathWrap
    checkMath(func_name, num_args, True, **args)
  File "/var/lib/jenkins/workspace/test/test_jit.py", line 6286, in checkMath
    cu = torch.jit.CompilationUnit(funcs_str)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/_script.py", line 1078, in __init__
    self.define(lang, _frames_up=_frames_up + 1)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/_script.py", line 1083, in define
    self._c.define(lang, rcb)
RuntimeError: replace_node->maybeOperator() INTERNAL ASSERT FAILED at "/var/lib/jenkins/workspace/torch/csrc/jit/passes/normalize_ops.cpp":38, please report a bug to PyTorch. invalid symbol replacement:140136
```

Which I've tried and failed to fix, even after asking internally.

**Comment by mruberry** [comment]
> The Windows CI builds have an internal compiler error, possibly related, but this PR isn't the source of the issue I think.
> 
> The Linux CI jobs fail with:
> 
> ```python
> Traceback (most recent call last):
>   File "/var/lib/jenkins/workspace/test/test_jit.py", line 6330, in test_math_ops
>     checkMathWrap(op, 1)
>   File "/var/lib/jenkins/workspace/test/test_jit.py", line 6256, in checkMathWrap
>     checkMath(func_name, num_args, True, **args)
>   File "/var/lib/jenkins/workspace/test/test_jit.py", line 6286, in checkMath
>     cu = torch.jit.CompilationUnit(funcs_str)
>   File "/opt/conda/lib/python3.6/site-packages/torch/jit/_script.py", line 1078, in __init__
>     self.define(lang, _frames_up=_frames_up + 1)
>   File "/opt/conda/lib/python3.6/site-packages/torch/jit/_script.py", line 1083, in define
>     self._c.define(lang, rcb)
> RuntimeError: replace_node->maybeOperator() INTERNAL ASSERT FAILED at "/var/lib/jenkins/workspace/torch/csrc/jit/passes/normalize_ops.cpp":38, please report a bug to PyTorch. invalid symbol replacement:140136
> ```
> 
> Which I've tried and failed to fix, even after asking internally.

Unfortunately Windows CI builds were broken for awhile. Sorry about that and the confusion it caused.

The Linux issue is more interesting. It turns out that the JIT already defines `degrees` and `radians` analogous to Python: 

https://github.com/pytorch/pytorch/blob/9c8021c0b110091451d361f47027715953bc3ecf/torch/jit/_builtins.py#L67

It does this because the JIT supports a subset of Python, so it has implemented many Python builtins.

This means that unconditionally converting from degrees to another name won't work, because the alias is attempting to convert these calls, too!

We could probably reverse the alias and have `rad2deg` alias to `degrees`, @eellison, what would you think of that? Although I don't want to send you on a wild goose chase, @hameerabbasi, so maybe a different issue would be more interesting until we resolve this one.

**Comment by eellison** [comment]
@mruberry I think we can add a check that the node has inputs and the first input is a Tensor before doing the alias replacement, that would fix this issue. Another fix would be to match on the full schema instead of just the kind in checking if there's an alias.

**Pull request closed without merging**